plugin-server/tests/worker/ingestion/person-state.test.ts

```diff
@@ -65,606 +68,1719 @@ describe('PersonState.update()', () => {
+                    id: uuid.toString(),
+                    properties: '{}',
+                    created_at: timestampch,
+                    version: 1,
+                    is_identified: 1,
+                })
+            )
+
+            // verify ClickHouse distinct_ids
+            await delayUntilEventIngested(fetchDistinctIdsRows)
+            const clickHouseDistinctIds = await hub.db.fetchDistinctIdValues(persons[0], Database.ClickHouse)
+            expect(clickHouseDistinctIds).toEqual(expect.arrayContaining(['old-user', 'new-user']))
+
+            // verify personContainer
+            expect(persons[0]).toEqual(await personContainer.get())
+        })
+
+        it('add distinct id while only anon person exists', async () => {
+            await hub.db.createPerson(timestamp, {}, {}, {}, 2, null, false, uuid.toString(), ['old-user'])
+
+            const personContainer = await personState({
+                event: '$identify',
+                distinct_id: 'new-user',
+                properties: {
+                    $anon_distinct_id: 'old-user',
+                },
+            }).update()
+            await hub.db.kafkaProducer.flush()
+
+            // verify Postgres persons
+            const persons = await hub.db.fetchPersons()
+            expect(persons.length).toEqual(1)
+            expect(persons[0]).toEqual(
+                expect.objectContaining({
+                    id: expect.any(Number),
+                    uuid: uuid.toString(),
+                    properties: {},
+                    created_at: timestamp,
+                    version: 1,
+                    is_identified: true,
+                })
+            )
+
+            // verify Postgres distinct_ids
+            const distinctIds = await hub.db.fetchDistinctIdValues(persons[0])
+            expect(distinctIds).toEqual(expect.arrayContaining(['old-user', 'new-user']))
+
+            // verify ClickHouse persons
+            const clickhousePersons = await delayUntilEventIngested(fetchPersonsRows)
+            expect(clickhousePersons.length).toEqual(1)
+            expect(clickhousePersons[0]).toEqual(
+                expect.objectContaining({
+                    id: uuid.toString(),
+                    properties: '{}',
+                    created_at: timestampch,
+                    version: 1,
+                    is_identified: 1,
+                })
+            )
+
+            // verify ClickHouse distinct_ids
+            await delayUntilEventIngested(fetchDistinctIdsRows)
+            const clickHouseDistinctIds = await hub.db.fetchDistinctIdValues(persons[0], Database.ClickHouse)
+            expect(clickHouseDistinctIds).toEqual(expect.arrayContaining(['old-user', 'new-user']))
+
+            // verify personContainer
+            expect(persons[0]).toEqual(await personContainer.get())
+        })
+
+        it('add distinct id while anon person does not exists with properties update', async () => {
```

Comment:
This test description is confusing as it's describing the end state but not the data going in with terminology not used anywhere in the code (anon person).

Given these tests are ~50 lines long it's really hard to find what's the important thing in this code it's critical we get descriptions correct.

```suggestion
        it('adds distinct_id and updates properties when $anon_distinct_id is passed and does not exist', async () => {
```

Reply:
I'll review this from perspective of jest output & can I understand both the pre & post conditions